### PR TITLE
don't overwrite distribution-scope label in dockerfile fix

### PR DESCRIPTION
**Pull request by rcerven**

### CODE REVIEW

File: tests/plugins/test_add_labels.py [hunk 1]
```diff
@@ -296,16 +296,30 @@ def test_add_labels_aliases(tmpdir, docker_tasker, caplog,
...
+    (None, None, "restricted", True),
+    ("public", None, "restricted", True),
+    ("private", None, "restricted", True),
+    ("restricted", 'public', "public", True),
+    ("restricted", 'restricted', "restricted", True),
+    ("restricted", 'private', "private", True)
 ])
-def test_dont_overwrite_distribution_scope(tmpdir, docker_tasker, labels):
+def test_dont_overwrite_distribution_scope(tmpdir, docker_tasker, parent_scope,
+                                           docker_scope, result_scope, overwrite):
```
Comment (twaugh) [inline tests/plugins/test_add_labels.py]: This all looks great, but the sense of `overwrite` seems to be inverted. I'd expect it to either be named `dont_overwrite` or have opposite True/False values.

File: tests/plugins/test_add_labels.py [hunk 2]
```diff
@@ -296,16 +296,30 @@ def test_add_labels_aliases(tmpdir, docker_tasker, caplog,
     if expected_log:
         assert expected_log in caplog.text()
 
-@pytest.mark.parametrize('labels', [{"distribution-scope": "restricted",
-                                     "dont_overwrite_if_in_dockerfile": "distribution-scope",
-                                     "auto_labels": [], 'aliases': {}},
-                                    {"distribution-scope": "restricted",
-                                     "auto_labels": [], 'aliases': {}}
+@pytest.mark.parametrize('parent_scope, docker_scope, result_scope, overwrite', [
```
Comment (vrutkovs) [inline tests/plugins/test_add_labels.py]: This param should be updated

File: tests/plugins/test_add_labels.py [hunk 3]
```diff
@@ -296,16 +296,30 @@ def test_add_labels_aliases(tmpdir, docker_tasker, caplog,
...
+    ("private", None, "restricted", False),
+    ("restricted", 'public', "public", False),
+    ("restricted", 'restricted', "restricted", False),
+    ("restricted", 'private', "private", False),
+    (None, None, "restricted", True),
+    ("public", None, "restricted", True),
+    ("private", None, "restricted", True),
+    ("restricted", 'public', "public", True),
+    ("restricted", 'restricted', "restricted", True),
+    ("restricted", 'private', "private", True)
```
Comment (twaugh) [inline tests/plugins/test_add_labels.py]: Minor nit: these are all using different quote types, double and single.

File: tests/plugins/test_add_labels.py [hunk 4]
```diff
@@ -318,16 +332,37 @@ def test_dont_overwrite_distribution_scope(tmpdir, docker_tasker, labels):
...
 
-    runner = PreBuildPluginsRunner(
-        docker_tasker,
-        workflow,
-        [{
-            'name': AddLabelsPlugin.key,
-            'args': {'labels': labels}
-        }]
-    )
+    if dont_overwrite:
```
Comment (vrutkovs) [inline tests/plugins/test_add_labels.py]: This large `if` can be minimized:
```
add_label_args = {
    'labels': {"distribution-scope": "restricted"},
    'auto_labels': [],
    'aliases': {},
}

if dont_overwrite:
    add_label_args['dont_overwrite_if_in_dockerfile'] = "distribution-scope"

runner = PreBuildPluginsRunner(
            docker_tasker,
            workflow,
            [{
                'name': AddLabelsPlugin.key,
                'args': add_label_args
            }]
        )
```

File: tests/plugins/test_add_labels.py [hunk 5]
```diff
@@ -296,16 +296,31 @@ def test_add_labels_aliases(tmpdir, docker_tasker, caplog,
...
+    ("restricted", 'public', "public", False),
+    ("restricted", 'restricted', "restricted", False),
+    ("restricted", 'private', "private", False),
+    (None, None, "restricted", True),
+    ("public", None, "restricted", True),
+    ("private", None, "restricted", True),
+    ("restricted", 'public', "public", True),
+    ("restricted", 'restricted', "restricted", True),
+    ("restricted", 'private', "private", True),
+    ("public", 'private', "private", True)
```
Comment (twaugh) [inline tests/plugins/test_add_labels.py]: Minor nit: quotation marks used inconsistently here.

File: tests/plugins/test_add_labels.py (hunk 4 above)
Comment (twaugh) [inline tests/plugins/test_add_labels.py]: Agreed. Let's not repeat things we don't need to.

File: tests/plugins/test_add_labels.py [hunk 6]
```diff
@@ -318,16 +333,24 @@ def test_dont_overwrite_distribution_scope(tmpdir, docker_tasker, labels):
     flexmock(workflow, base_image_inspect=labels_conf_base)
     setattr(workflow.builder, 'df_path', df.dockerfile_path)
 
+    wf_args = {
+        'labels': {"distribution-scope": "restricted"},
+        'auto_labels': [],
+        'aliases': {},
+    }
+    if dont_overwrite:
+        wf_args["dont_overwrite_if_in_dockerfile"] = "distribution-scope"
```
Comment (twaugh) [inline tests/plugins/test_add_labels.py]: Should this be an iterable?

**Comment by rcerven** [comment]
it is tuple

**Pull request merged**